packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2582,6 +2582,10 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             }
         }
 
+        if (this.deltaManager.readOnlyInfo.readonly) {
```

Comment:
You will hit a merge here - there is an interesting question to be made (after my refactoring) - should this check stay in submit() or move to flush()? I think it should stay here, even though there is an obvious race - we can move to readonly before flush() promise runs, though I think it would point out to user error (i.e. batches are supposed to be synchronous, so if this race happens, it points to somebody change container state and calling forceReadonly() in synchronous flow.